utils/stall_free.hh

```diff
@@ -66,5 +66,16 @@ void stall_free_clear(std::list<T>& list) {
     }
 }
 
+template<class T>
+future<> stall_free_clear_outside_thread(std::list<T>& list) {
+    return repeat([&list] () mutable {
```

Comment:
No need for "outside_thread" if it returns a future.

Also, better to pass by rvalue reference.

Reply:
I can drop the stall_free_clear version, so we can have a single version that returns a future.